### PR TITLE
feat(volar): improve Volar support (wip)

### PR DESCRIPTION
**Pull request by Tahul**

### CODE REVIEW

File: src/volar.ts [hunk 1]
```diff
@@ -43,68 +23,21 @@ const plugin: VueLanguagePlugin = _ => ({
...
+      embeddedFile.parentFileName = fileName
+      embeddedFile.content.push(...content)
       embeddedFile.content.push([
-        cssMatches[0],
+        style.content,
         style.name,
-        cssMatches.index,
-        fullCapabilities,
+        0,
+        FileCapabilities.full,
```
Comment (johnsoncodehk) [inline src/volar.ts:37]: It should be `FileRangeCapabilities.full` instead.

File: src/volar.ts [hunk 2]
```diff
@@ -1,40 +1,20 @@
...
-        match,
-        vueTag,
-        index,
-        fullCapabilities,
-      ])
-      embeddedFile.content.push('\n')
-    }
-
-    // Handle <vue> files
-    if (embeddedFile.fileName.replace(fileName, '').match(/^\.(js|ts|jsx|tsx)$/)) {
```
Comment (johnsoncodehk) [inline src/volar.ts:28]: You should add this back and add `variants` virtual code for script block.

File: playground/tsconfig.json [hunk 3]
```diff
@@ -104,7 +104,7 @@
   "include": [
     "./.nuxt/nuxt.d.ts",
     "../**/*",
-    "../../**/*"
+    "../**/*"
```
Comment (johnsoncodehk) [inline playground/tsconfig.json:107]: it seems still including unneeded files from dist.

File: src/transforms/vue/variants.ts [hunk 4]
```diff
@@ -74,12 +74,12 @@ export function resolveVariantsProps(variants, isTs: boolean) {
 
       const isBooleanVariant = Object.keys(variant).some(key => (key === 'true' || key === 'false'))
       if (isBooleanVariant) {
-        prop.type = isTs ? ' [Boolean, Object] as PropType<boolean | { [key in PinceauMediaQueries]?: boolean }>' : ' [Boolean, Object]'
+        prop.type = isTs ? ' [Boolean, Object] as PropType<boolean | { [key in import(\'pinceau\').PinceauMediaQueries]?: boolean }>' : ' [Boolean, Object]'
```
Comment (johnsoncodehk) [inline src/transforms/vue/variants.ts:77]: Note:
- `PropType` -> `import('vue').PropType`
- Add `as const` for `type` and `default.`

**Comment by Tahul** [comment]
Thank you so much @johnsoncodehk 🙏

**Pull request merged**